extensions/amp-story/0.1/amp-story.js

```diff
@@ -1004,6 +997,36 @@ export class AmpStory extends AMP.BaseElement {
   }
 
 
+  /** @private */
+  onCurrentPageIdUpdate_(pageId) {
+    this.setHistoryState_(pageId);
+  }
+
+
+  /**
+   * Save page id using history API
+   * @private
+   */
+  setHistoryState_(pageId) {
+    const history = this.win.history;
+    if (history.replaceState && this.getHistoryStatePage_() !== pageId) {
```

Comment:
The store should guarantee that the callback isn't called unless the value has changed, but I'm fine with being extra cautious

Reply:
the problem is when leaving the story to an outlink, then upon return the subscriber is called